ddp-workspace/projects/ddp-singular/src/app/components/activities-list/activities-list.component.html

```diff
@@ -78,13 +78,13 @@
           <ng-container *ngTemplateOutlet="isActivityEditable(activity) ? update : view"></ng-container>
 
           <ng-template #update>
-            <button type="button" class="button button--outline button--sm fixed-button" (click)="onEditClick(activity)">
+            <button type="button" class="button button--outline button--sm fixed-button" (click)="onEditClick(activity)"  [disabled]="disableButtons">
               {{ (isMedicalRecordFileUpload(activity) ? 'SDK.UploadButton' : 'SDK.EditButton') | translate }}
             </button>
           </ng-template>
 
           <ng-template #view>
-            <button type="button" class="button button--outline button--sm fixed-button" (click)="onViewClick(activity)">
+            <button type="button" class="button button--outline button--sm fixed-button" (click)="onViewClick(activity)"  [disabled]="disableButtons">
               {{ 'SDK.ReviewButton' | translate }}
             </button>
```

Comment:
To be triple sure that we are not allowing double clicks or duplicate requests, I am disabling buttons whenever something is going on.